### PR TITLE
Compile time options for HIDE_SOCKET_FILENO and FAST_CONNECT_DROP

### PR DESCRIPTION
**Pull request by noway**
Enables to hide Socket.fileno function (from external SDK for example, so it wouldn't be possible to abuse the underlying Socket connection instead of using intercept callback properly)

As well as fast Connect Drop where it can return None instead of the event on disconnect of the peer.

Enabled through compile time options in setup.py.

Tested it with py2k and py3k, works fine.

Voice the concern if you have any :-)

### CODE REVIEW

**Comment by NotAFile** [comment]
@aresch could you have a look at this?

File: enet.pyx [hunk 1]
```diff
@@ -925,15 +928,31 @@ cdef class Host:
         the host and its peers. The timeout is in milliseconds.
         """
 
+        cdef int number_result
```
Comment (aresch) [inline enet.pyx]: Why is this necessary? And if it is, could it simply be renamed to 'result' and used in both cases?

File: enet.pyx [hunk 2]
```diff
@@ -925,15 +928,31 @@ cdef class Host:
...
-                raise IOError("Servicing error - probably disconnected.")
+            global FAST_CONNECT_DROP
+
+            if FAST_CONNECT_DROP == 1:
+                number_result = enet_host_service(
+                    self._enet_host, &(<Event> event)._enet_event, timeout)
+
+                if number_result < 0:
+                    raise IOError("Servicing error - probably disconnected.")
+                elif number_result == 0:
```
Comment (aresch) [inline enet.pyx]: I think this could be simplified to:

elif result == 0 and FAST_CONNECT_DROP:

Comment (aresch) [inline enet.pyx]: I think this could be simplified.

elif result == 0 and FAST_CONNECT_DROP:

File: enet.pyx (hunk 1 above)
Comment (aresch) [inline enet.pyx]: Why is this necessary? If it is, change it to 'result'.

**Comment by aresch** [comment]
I'm a little confused on the point of hiding the fileno() method, but I'd be more inclined to just remove it then trying to add a compile option to hide it since I'm not convinced its particularly useful.  Also, I'm not sure why the FAST_CONNECT_DROP needs to be a compile-time option. It could probably just be added as an kwarg to Host.__init__ or even just to the service() method.  I think that doing this would reduce the complexity of this change.

**Comment by noway** [comment]
HIDE_SOCKET_FILENO is indeed pretty redundant and I would go without it. I'd simply forego this option as because it is pretty useless. 

Re: kwargs to `Host.__init__`: sounds like a great idea, definitely something I'd go with.

**Comment by noway** [comment]
Superseded by #9
